ui/packages/catalog-ui-search/src/main/webapp/js/model/List.js

```diff
@@ -86,6 +112,47 @@ module.exports = Backbone.AssociatedModel.extend(
         this.updateQuery
       )
     },
+    set(data, ...args) {
+      if (typeof data === 'object') {
+        // for backwards compatability
+        if (data['list.cql'] && !data['list.filters']) {
+          try {
+            const filterTree = CQLUtils.transformCQLToFilter(data['list.cql'])
+            data['list.filters'] = simplifyListFilter(filterTree)
+          } catch (e) {
+            console.log('Invalid cql: ' + data['list.cql'])
+          }
+          data['list.cql'] = undefined
+        } else if (
+          data['list.filters'] &&
+          typeof data['list.filters'] === 'string'
+        ) {
+          try {
+            data['list.filters'] = JSON.parse(data['list.filters'])
+          } catch (e) {
+            data['list.filters'] = undefined
+          }
+        }
+      } else if (data === 'list.filters' && args.length == 1) {
+        args = [simplifyListFilter(args[0])]
+      }
+      if (!IGNORED_DEPRECATED_ATTRIBUTES.includes(data)) {
+        return Backbone.AssociatedModel.prototype.set.call(this, data, ...args)
+      }
+    },
+    get(attr) {
+      if (!IGNORED_DEPRECATED_ATTRIBUTES.includes(attr)) {
+        return Backbone.AssociatedModel.prototype.get.call(this, attr)
+      }
+    },
+    toJSON(...args) {
+      const json = Backbone.AssociatedModel.prototype.toJSON.call(this, ...args)
+      const keepObjs = args.length > 0 ? args.includes('keepObjs') : false
+      if (!keepObjs && typeof json['list.filters'] === 'object') {
+        json['list.filters'] = JSON.stringify(json['list.filters'])
```

Comment:
In some cases such as here https://github.com/codice/ddf/blob/3378b5a380422e1d7747bb47ec8e16bfedaf8b5d/ui/packages/catalog-ui-search/src/main/webapp/component/result-add/result-add.view.js#L90 we want to iterate through the list.filters as an object and have no use of its stringify version. I could get it as a string, but will need to convert it as an object to iterate through it, thought this would be more efficient since won't need to do 2 conversions. Should I change it so each time want to use it as an object, the person calling it, should be responsible for converting it?

Reply:
I think we can store it as an object by default